include/zephyr/drivers/misc/timeaware_gpio/timeaware_gpio.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for Timed GPIO drivers
+ */
+#ifndef ZEPHYR_DRIVERS_MISC_TIMEAWARE_GPIO_TIMEAWARE_GPIO
+#define ZEPHYR_DRIVERS_MISC_TIMEAWARE_GPIO_TIMEAWARE_GPIO
+
+/**
+ * @brief Timed GPIO Interface
+ * @defgroup tagpio_interface Timed GPIO Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/slist.h>
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <zephyr/device.h>
+#include <zephyr/syscall_handler.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Event polarity
+ */
+enum tagpio_pin_polarity {
```

Comment:
this will come across as a nit, but it would appear that the most common acronym for time-aware GPIO is TGPIO, not TAGPIO (in Linux kernel at leasrt), so I think it's worth reconsidering if TAGPIO is appropriate?
In any case, the use of TGPIO vs. TAGPIO is not consistent throughout the PR :)

Reply:
Changed from TAGPIO to TGPIO all over the PR